### PR TITLE
python-bareos: extends the documentation

### PR DESCRIPTION
**Pull request by joergsteffens**
This PR contains

- some general DeveloperGuide fixes,
- add a note about Python docstring formating,
- updates python-bareos  README file and its docstrings,
- add python-bareos to DeveloperGuide (by extracting doctrings via Sphinx extension autosummary).

### CODE REVIEW

**Review by arogge — CHANGES_REQUESTED** [review]
The content looks great overall, some of the commits were in 2021 and need a more recent year in the copyright notice. Also the integration of the autosummary into the hierarchy is broken (i.e. it is located under "Exmaple")

File: docs/manuals/source/DeveloperGuide/api.rst [hunk 1]
```diff
@@ -38,31 +38,28 @@ ways that Bareos is designed to facilitate this:
    than data.
 
 -  The Director has a number of commands that are non-interactive. They
-   all begin with a period, and provide things such as the list of all
+   all begin with a period and provide things such as the list of all
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply docs/manuals/source/DeveloperGuide/api.rst:41]: We decided, that comma before "and" and "or" are not required.

File: docs/manuals/source/DeveloperGuide/PythonBareos.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   bareos.util.path
+   bareos.exceptions
+
+
+..
+
+   Skip deprecated modules
+
+   bareos.bsock.bsock
+   bareos.bsock.bsockjson
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply docs/manuals/source/DeveloperGuide/PythonBareos.rst:78]: This is a commented section. See ".. " above

[the remark this answers is not in the document]
Comment (joergsteffens) [reply docs/manuals/source/DeveloperGuide/PythonBareos.rst:76]: I see, the comment marker misses the trailing space. I fixed this in additional commit.

File: docs/manuals/source/DeveloperGuide/PythonBareos.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+..
+   autosummary::
+   :toctree: ../include/autogenerated/autosummary/python-bareos
+   :caption: python-bareos
+   :template: custom-module-template.rst
+   :recursive:
+
+   bareos
+
+.. autosummary::
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply docs/manuals/source/DeveloperGuide/PythonBareos.rst:49]: I modified it.

**Review by arogge — APPROVED** [review]

**Pull request merged**